android/app/src/main/kotlin/coop/polypoly/polypod/polyOut/PolyOut.kt

```diff
@@ -103,29 +103,47 @@ open class PolyOut(
 
     open suspend fun readdir(
         id: String
-    ): Array<String> {
-
+    ): Array<Map<String, String>> {
         val fs = Preferences.getFileSystem(context)
+        val retList = mutableListOf(mapOf<String,String>())
+        
+        retList.removeFirst()
         if (id == "") {
             val newFs = fs.filter {
                 File(idToPath(it.key, context)).exists()
             }
             Preferences.setFileSystem(context, newFs)
-            return newFs.keys.toTypedArray()
+
+            val allIds = newFs.keys;
+            
+            for (idPath in allIds){
+                val relPath = getRelativePathFromId(idPath)
```

Comment:
We're listing the root directory here how I understand the existing code. By definition, `idPath` should be identical to `relPath` then.

Reply:
yes absolutely, that is true and I created a case inside of `getRelativePathFromId `to work with this condition as to keep convention of using `getRelativePathFromId`, but I agree it is better to just use it since it is there :)